### DISCUSSION TITLE
Android app: Where should I put detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:VERSION")?

### DISCUSSION THREAD

**Discussion by VincentJoshuaET** [detekt/detekt#4092] [Q&A]
Should I only put it in the root project, `allprojects`, or `subprojects`?

```
dependencies {
    `detekt-formatting`
}

tasks.withType<Detekt> {
    jvmTarget = JavaVersion.VERSION_11.toString()
}
```

```
subprojects {
    dependencies {
        `detekt-formatting`
    }

    tasks.withType<Detekt> {
        jvmTarget = JavaVersion.VERSION_11.toString()
    }
}
```

```
allprojects {
    dependencies {
        `detekt-formatting`
    }

    tasks.withType<Detekt> {
        jvmTarget = JavaVersion.VERSION_11.toString()
    }
}
```

Also, where should I put `tasks.withType<Detekt>().jvmTarget`?

**Comment by cortinico** [comment]
> Should I only put it in the root project, `allprojects`, or `subprojects`?

You should put them at the same level where you apply the Detekt Gradle plugin.
If you apply the plugin inside the `allprojects`, or `subprojects` block, then you put it there.

I don't seen from your snippets how you apply the Detekt plugin, therefore I can't advice you further on where to place it.
